### PR TITLE
ci(spread): add reusable workflow for Spread tests

### PR DESCRIPTION
**Pull request by cjdcordeiro**
# Proposed changes

This PR introduces a new reusable GitHub workflow for running Spread tests. 

Each branch will have its own `spread.yaml` configuration, and this workflow will be called from the branches, on every PR. 

The tests shall only run for:
 - the SDFs which are added or modified
 - the tests which are added or modified 

## Testing

Tested the execution of this workflow here:
 - https://github.com/cjdcordeiro/chisel-releases/pull/11 (with additional mock commits to ensure the logic works when filtering the added and modified files)
 - https://github.com/cjdcordeiro/chisel-releases/pull/13

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

**Review by letFunny — APPROVED** [review]
Thanks Cris, it is looking very good! From the linked PRs I am missing the use case where the spread tests already exists and changing the SDF triggers the integration test (i.e. `changed-slices` triggering instead of `changed-tests`).

**Comment by cjdcordeiro** [comment]
> Thanks Cris, it is looking very good! From the linked PRs I am missing the use case where the spread tests already exists and changing the SDF triggers the integration test (i.e. `changed-slices` triggering instead of `changed-tests`).

Thanks.

As for that use case, it's this one here: https://github.com/cjdcordeiro/chisel-releases/actions/runs/9905003363/job/27363593631?pr=11#step:8:27

From the logs you can see the SDF being considered as a trigger for that test. It just so happens that the test itself is also a trigger, cause it is also being created for the first time.

File: .github/workflows/spread.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        id: changed-slices
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            slices:
+              - added|modified: 'slices/**/*.yaml'
+          list-files: shell
+      
+      - name: Check changed test directories
+        uses: tj-actions/changed-files@v35
```
Comment (rebornplusplus) [inline .github/workflows/spread.yaml:27]: Can't we use the `dorny/paths-filter` action from above here too?

**Review by rebornplusplus — APPROVED** [review]
Looks good to me, thank you! 

Just one quick question about using less distinct actions.

File: .github/workflows/spread.yaml (hunk 1 above)
Comment (rebornplusplus) [reply .github/workflows/spread.yaml:27]: Ah okay, thought so.
No worries, approved it already. :+1:

**Pull request merged**